src/Simplex/Chat/Remote.hs

```diff
@@ -185,9 +191,20 @@ relayCommand http s =
 
 -- | Convert swift single-field sum encoding into tagged/discriminator-field
 owsf2tagged :: J.Value -> J.Value
-owsf2tagged = \case
-  J.Object todo'convert -> J.Object todo'convert
-  skip -> skip
+owsf2tagged val = case val of
+  J.Object o -> J.Object $ case JM.toList o of
+    [OwsfTag, (k, v)] -> tagged k v
+    [(k, v), OwsfTag] -> tagged k v
+    ps -> JM.fromList $ map (second owsf2tagged) ps
```

Comment:
Structure-preserving transforms can benefit from `KeyMap` being a funtor on values: `fmap owsf2tagged o`.

Reply:
`tagged` terminates recursion, screening off OWSFs from the conversion.

`tagged k (owsf2tagged v)` ?